### DISCUSSION TITLE
Rework the Cache system to rely on Symfony Cache

### DISCUSSION THREAD

**Discussion by DarkGhostHunter** [laravel/framework#50584] [Ideas]
I firmly believe that, now that Laravel 11 has shipped, work should be done to _rework_ the cache system of Laravel 12 to rely on Symfony Cache rather than own implementations, especially on the Redis side of things.

There are some great things about leaning a little more on Symfony Cache.

- [Cache Items and Tags are detached](https://symfony.com/doc/current/components/cache/cache_invalidation.html#tag-aware-adapters). You can store big items in files, and use redis solely for tagging them.
- [Stampede prevention](https://symfony.com/doc/current/components/cache.html#stampede-prevention) with [probabilistic early-expiration](https://en.wikipedia.org/wiki/Cache_stampede#Probabilistic_early_expiration).
- [Custom Marshalling and serialization](https://symfony.com/doc/current/components/cache.html#marshalling-serializing-data). That means the dev could _sign_ the serialization to avoid tampering.
- [Programmatic prune](https://symfony.com/doc/current/components/cache/cache_pools.html#component-cache-cache-pool-prune). For example, the dev can change the probability to prune entries after a cache operation. 
- [Chained cache stores](https://symfony.com/doc/current/components/cache/adapters/chain_adapter.html). Mix and match multiple caches.
- [Redis tags adapter](https://symfony.com/doc/current/components/cache/adapters/redis_adapter.html#working-with-tags).

I honestly don't think the extent of the changes, but I think it would make the cache system easier to maintain.

**Comment by dam1r89** [comment]
This is a code sample to get you started if you want to use it now.
And of course, install Symfony cache package first.

```
// put it in AppServiceProvider for example
$this->app->singleton(\Symfony\Contracts\Cache\TagAwareCacheInterface::class, function () {
    $redisAdapter = new RedisAdapter(
        $this->app->make('redis')
            ->connection(config('cache.stores.redis.connection'))
            ->client()
    );

    return new TagAwareAdapter($redisAdapter);
});

/**
 * @var $cache \Symfony\Contracts\Cache\TagAwareCacheInterface&\Psr\Cache\CacheItemPoolInterface
 */
$cache = app(\Symfony\Contracts\Cache\TagAwareCacheInterface::class);
```
